lolcation.js

```diff
@@ -36,9 +38,15 @@ module.exports = function lolcation(loc) {
   } else if ('string' === type) {
     finaldestination = new URL(loc, {});
     for (key in ignore) delete finaldestination[key];
-  } else if ('object' === type) for (key in loc) {
-    if (key in ignore) continue;
-    finaldestination[key] = loc[key];
+  } else if ('object' === type) {
+    for (key in loc) {
+      if (key in ignore) continue;
+      finaldestination[key] = loc[key];
+    }
+
+    if (finaldestination.slashes === undefined) {
+      finaldestination.slashes = slashes.test(loc.href);
```

Comment:
We can reuse [`extractProtocol`](https://github.com/unshiftio/url-parse/blob/f30e236d4f39b1f8ee47f6fc29564944e65a721a/index.js#L46-L53) here, but this is faster.